include/lsst/meas/astrom/astrometry_net.h

```diff
@@ -0,0 +1,237 @@
+    @param[in] uniqueIds  if true then only return unique IDs (the first of each seen)
+
+    Returned schema:
+    - id
+    - coord: sky position (an lsst::afw::coord::IcrsCoord)
+    - centroid: centroid on some exposure, if relevant (an lsst::afw::geom::Point2D);
+        returned value is not set
+    - hasCentroid: if true then centroid has been set; returned value is false
+    - <filterName>_flux: flux in the specified filter (double)
+    - <filterName>_fluxSigma: flux uncertainty in the specified filter (double)
+    - resolved (if starGalCol specified): true if object is not resolved
+    - variable (if varCol specified): true if brightness is variable
+    - photometric: true if not resolved (or starGalCol blank) and not variable (or varCol blank);
+        note that if starGalCol and varCol both blank then all objects are claimed to be photometric
+    */
+    lsst::afw::table::SimpleCatalog getCatalog(
+        std::vector<index_t*> inds,
+        lsst::afw::coord::Coord const &ctrCoord,
+        lsst::afw::geom::Angle const &radius,
+        const char* idCol,
+        std::vector<std::string> const& filterNameList,
+        std::vector<std::string> const& magColList,
+        std::vector<std::string> const& magErrColList,
+        const char* starGalCol,
+        const char* varCol,
+        bool uniqueIds=true);
+
+    std::shared_ptr<lsst::daf::base::PropertyList> getSolveStats() const;
+
+    std::shared_ptr<lsst::afw::image::Wcs> getWcs();
+
+    bool didSolve() const {
+        return solver_did_solve(_solver.get());
+    }
+
+    void run(double cpulimit);
+
+    std::pair<double, double> getQuadSizeRangeArcsec() const {
+        double qlo,qhi;
+        solver_get_quad_size_range_arcsec(_solver.get(), &qlo, &qhi);
+        return std::make_pair(qlo, qhi);
+    }
+
+    /**
+     * Add indices to the solver
+     *
+     * The indices are bare pointers whose memory is managed by the caller.
+     * Typically the indices are owned by a MultiIndex object owned by the caller.
+     */
+    void addIndices(std::vector<index_t*> inds);
+
+    /**
+     * Set parity to flipped (if true) or normal (if false)
+     */
+    void setParity(bool flipped) {
+        _solver->parity = flipped ? PARITY_FLIP : PARITY_NORMAL;
+    }
+
+    void setMatchThreshold(double threshold) {
+        solver_set_keep_logodds(_solver.get(), threshold);
+    }
+
+    void setPixelScaleRange(double low, double high) {
+        _solver->funits_lower = low;
+        _solver->funits_upper = high;
+    }
+
+    void setRaDecRadius(double ra, double dec, double radius_deg) {
+        solver_set_radec(_solver.get(), ra, dec, radius_deg);
+    }
+
+    void setImageSize(int width, int height);
+
+    void setMaxStars(int maxStars) {
+        _solver->endobj = maxStars;
+    }
+
+    void setStars(lsst::afw::table::SourceCatalog const & srcs, int x0, int y0);
+
+private:
+
+    struct _Deleter {
```

Comment:
I don't know. It is a private local class and I wanted to make it clear that nobody else can use it.